src/clr-angular/forms/datepicker/providers/date-form-control.service.ts

```diff
@@ -30,6 +31,24 @@ export class DateFormControlService {
     this._dirtyChange.next();
   }
 
+  /**
+   * Handle Date value across components
+   */
+  private dataClearEvent: Subject<void> = new Subject();
```

Comment:
Perhaps I am missing something here. Is there a reason this service shouldn't use a private value to store the current date value for the datepicker (on this service) and use the typical `private property/Subject/Observable` pattern? e.g

```
private currentDate: Date | string;
private _dateValue: Subject<Date|string> = new Subject<Date|string>();
get dateValue: Observable<Date|string>() { return this._dateValue.asObservable(); }
```

Then this service would also centralize the `setter for the date value` and `clear` functions for currentDate which would allow listeners to reactively get the updates/changes to the dateValue.